libnetdata/socket/security.c

```diff
@@ -49,32 +53,59 @@ void security_openssl_library()
 #endif
 }
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+/**
+ * TLS version
+ *
+ * Returns the TLS version depending of the user input.
+ *
+ * @param lversion is the user input.
+ *
+ * @return it returns the version number.
+ */
+int tls_select_version(const char *lversion) {
+    if (!strcmp(lversion, "1.1"))
+        return TLS1_1_VERSION;
+    else if (!strcmp(lversion, "1.2"))
+        return TLS1_2_VERSION;
+#if OPENSSL_VERSION_NUMBER >= 0x1010101bL
+    else if (!strcmp(lversion, "1.3"))
+        return TLS1_3_VERSION;
+#endif
+
+    return TLS_MAX_VERSION;
```

Comment:
Is `TLSv1` a thing. I see i took it into account adding SSL protocl parsing in weblog

https://github.com/netdata/go.d.plugin/blob/1e6132f9572373005ab13ecdba9ef56a6bb6ca8b/modules/weblog/logline.go#L566-L575

i dont remember details tho 😄

Reply:
Considering that you added, I will also bring. For sure it is not the safest option, but people can use it yet.

Reply:
I have a second thought, it makes sense for weblog because it has to parse logs, but i think it doesnt make sense for Netdata to have it as an option.

Sorry for the noise 😅

Reply:
he last commit brought the missing TLS versions, but SSL is too dangerous, so I will keep it out. I updated the documentation to give better understanding about what means the new option `tls version`.

Reply:
> Sorry for the noise sweat_smile

Do not worry, I brought all TLS  options, because I think it is good to keep compatibility with old versions, but we also need to consider the security with old versions, so I think we do not need to extend so much to the past!

Thank you for your contribution! :)